### PR TITLE
Explosion inhibitor

### PR DESCRIPTION
**Pull request by AnturK**
:cl:
add: enviro-kinetic dampeners - machines weakening explosions can now be researched and constructed.
/:cl:

Machine that significantly weakens explosions starting within it range and lowers the effect on turfs within it's range from far off explosions. Can be unanchored and moved around but requires 30s calibration before it starts working again.

TODO:
- [x]  ~Needs a sprite.~ Basic sprite thanks to @trollbreeder 
- [ ] Construction and research cost and effect strength up to discussion.
- [x] Code cleanup

### CODE REVIEW

**Comment by trollbreeder** [comment]
Codersprites by me. They're probably terrible but they're workable
image

**Comment by XDTM** [comment]
I'm kinda wondering if it will be useful for the crew, given that it's in a pretty advanced tech node, and traitors have been known to get maxcaps far earlier than it's possible to research it, much less set it up.

**Comment by evilcyborgmrt** [comment]
Explosion inhibitor sums it up but the name is a bit dry. Maybe give it a more sciencey name like percussive dampener or something?

**Comment by AnturK** [comment]
The research requirements and cost can always be lowered, this is just first iteration.

I'm open to any more flavourful names, but it needs to suggest what it actually does.

**Comment by Shadowflame909** [comment]
How about Chronostasis Explosifier

**Comment by XDTM** [comment]
that seems like putting time bomb through a thesaurus

**Comment by Shadowflame909** [comment]
Okay Chi Delta Tau Mu

**Comment by AnturK** [comment]
Okay, new bit more fluffy name added. Power is now variable so it's easy to balance later.

**Comment by kevinz000** [comment]
🎉

File: code/__DEFINES/dcs/signals.dm [hunk 1]
```diff
@@ -442,3 +444,10 @@
 #define COMSIG_XENO_TURF_CLICK_SHIFT "xeno_turf_click_shift"				//from turf ShiftClickOn(): (/mob)
 #define COMSIG_XENO_TURF_CLICK_CTRL "xeno_turf_click_alt"					//from turf AltClickOn(): (/mob)
 #define COMSIG_XENO_MONKEY_CLICK_CTRL "xeno_monkey_click_ctrl"				//from monkey CtrlClickOn(): (/mob)
+
+#define COMSIG_EXPLOSION_TURF_BEFORE_EX_ACT "explosion_turf_before_ex_act"
+	#define COMPONENT_EXPLOSION_SKIP_TURF 1
```
Comment (Rohesie) [inline code/__DEFINES/dcs/signals.dm:449]:
```suggestion
	#define COMPONENT_EXPLOSION_SKIP_TURF (1<<0)
```
Returns are always flags, so flag notation to make it clear is positive.

File: code/__DEFINES/dcs/signals.dm [hunk 2]
```diff
@@ -442,3 +444,10 @@
...
+	#define COMPONENT_EXPLOSION_MODIFY 2
```
Comment (Rohesie) [inline code/__DEFINES/dcs/signals.dm:450]:
```suggestion
	#define COMPONENT_EXPLOSION_MODIFY (1<<1)
```

File: code/game/machinery/explosion_inhibitor.dm [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	. = ..()
+	if(state != INHIB_OFF && machine_stat & NOPOWER)
+		visible_message("<span class='warning'>[src] shuts down.</span>")
+		deactivate()
+
+/obj/machinery/explosion_inhibitor/attackby(obj/item/I, mob/living/user, params)
+	if(default_unfasten_wrench(user, I))
+		power_change()
+		return
+	return ..()
```
Comment (Rohesie) [inline code/game/machinery/explosion_inhibitor.dm:61]:
```suggestion
/obj/machinery/explosion_inhibitor/wrench_act(mob/living/user, obj/item/I)
	if(default_unfasten_wrench(user, I))
		power_change()
		return TRUE
```

**Comment by ExcessiveUseOfCobblestone** [comment]
Re-PR once issues are addressed.

**Pull request closed without merging**